app/src/main/java/com/spendster/presentation/homeScreen/HomeActivity.java

```diff
@@ -77,4 +81,9 @@ private boolean loadFragment(Fragment fragment){
         }
         return false;
     }
+
+    public void setToolbarCaption(int caption){
+        TextView toolbarCaption = findViewById(R.id.toolbarCaption);
```

Comment:
@SimonMaha you should do it once. Currently, you are creating a view each on calling this method